solidity/random-beacon/contracts/RandomBeacon.sol

```diff
@@ -562,6 +647,59 @@ contract RandomBeacon is Ownable {
         }
     }
 
+    /// @notice Punishes the given operators by kicking them out from the
+    ///         sortition pool and blocking their reinsertion for a given period
+    ///         of time.
+    /// @dev By the way, this function releases gas deposits made by operators
+    ///      during their registration. See `registerOperator` function. This
+    ///      action makes punishments cheaper gas-wise.
+    /// @param ids IDs of punished operators.
+    /// @param punishmentDuration Duration of the punishment period in seconds.
+    ///
+    /// TODO: This method can be probably optimized in future (depends on
+    ///       changes in the sortition pool). Until then we need to make it
+    ///       that way because we need to make sure we remove operators which
+    ///       are actually in the pool to avoid revert. Two loops are needed
+    ///       because Solidity doesn't allow for memory arrays and mappings.
+    function punishOperators(uint32[] memory ids, uint256 punishmentDuration)
+        internal
+    {
+        address[] memory operators = sortitionPool.getIDOperators(ids);
+        uint256 operatorsInPoolCount = 0;
+
+        for (uint256 i = 0; i < operators.length; i++) {
+            if (sortitionPool.isOperatorInPool(operators[i])) {
+                operatorsInPoolCount++;
+            }
+        }
+
+        uint32[] memory operatorsInPoolIDs = new uint32[](operatorsInPoolCount);
+        uint256 j = 0;
+
+        for (uint256 i = 0; i < operators.length; i++) {
+            address operator = operators[i];
+
+            // Set the punishment regardless the operator is actually
+            // registered in the sortition pool. If the operator is not in pool
+            // at the moment, the punishment will prevent they to join during
```

Comment:
prevent they to join -> prevent them from joining

Reply:
See https://github.com/keep-network/keep-core/pull/2671/commits/a400d36b149453ad8ffae6a8d739236f50841758